tcbuilder/backend/build.py

```diff
@@ -228,6 +229,44 @@ def make_download_fname(fname):
 
     return out_fname, is_temp
 
+def _load_config_file_yaml(config_path):
+    """
+    Load the YAML configuration file
+    """
+
+    # Load the YAML configuration file (user-supplied):
+    with open(config_path) as file:
+        try:
+            config = yaml.load(file, Loader=yaml.FullLoader)
+            return config
+
+        except yaml.YAMLError as ex:
+            error_exc = ParseError(getattr(ex, "problem", "parsing error"))
+            error_exc.set_source(file=config_path)
+            if hasattr(ex, "problem_mark"):
+                mark = getattr(ex, "problem_mark")
+                error_exc.set_source(line=mark.line, column=mark.column)
+            raise error_exc
+
+
+def _load_config_file_json(config_path):
+    """
+    Load the JSON configuration file
+    """
+
+    # Load the JSON configuration file (user-supplied):
+    with open(config_path) as file:
+        try:
+            config = json.load(file)
+            return config
+
+        except json.JSONDecodeError as ex:
+            error_exc = ParseError(getattr(ex, "msg", "parsing error"))
+            error_exc.set_source(file=config_path)
+            if hasattr(ex, "doc"):
+                error_exc.set_source(line=ex.doc)
```

Comment:
Why does this use `ex.doc` as the line number? Seems that `JSONDecodeError` has `lineno` (and also `colno`) attributes: https://docs.python.org/3/library/json.html#json.JSONDecodeError